xlml/utils/startup_script_util.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def genereate_startup_script(main_command: str) -> str:
+  return f"""
+bash -c '{main_command} 2>&1 | tee /tmp/logs &
```

Comment:
I have the closing `'` at line 23.

Here is the main command I used:
https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/83/files#diff-c697bd6d7e4d4647f9a1e3e094ded8e445df4a82583823b7280e8f90154e3925R54-R62

Reply:
Oh I see, let's use something more like `bash -c '{main_command}' 2>&1 | ...`, otherwise the redirect and tee will only apply to the last statement in the training script (e.g. a script of `echo hello; echo world` will turn into `echo hello; echo world 2>&1 | tee /tmp/logs` when we really want the entire `bash -c` to be redirected/tee'd)

The command linked is probably working out because the `'` don't contain whitespace - in general, it's best to escape the embedded commands.

Reply:
Done. Thanks for the tip!